rust/stub/src/pe_loader.rs

```diff
@@ -151,11 +151,10 @@ impl Image {
         handle: Handle,
         system_table: &SystemTable<Boot>,
         load_options: &CStr16,
-    ) -> Status {
+    ) -> uefi::Result<()> {
         let mut loaded_image = system_table
             .boot_services()
-            .open_protocol_exclusive::<LoadedImage>(handle)
-            .expect("Failed to open the LoadedImage protocol");
+            .open_protocol_exclusive::<LoadedImage>(handle)?;
```

Comment:
When the UEFI fails to open `LoadedImage` on our image handle, things are extremely messed up already. I'm not sure if there is any point in even bothering to try to return to the bootloader, which is likely not going to be able to work properly either.